skaha/src/main/java/org/opencadc/skaha/session/PostAction.java

```diff
@@ -572,7 +576,8 @@ public void attachSoftware(String image, List<String> params) throws Exception {
                     log.debug("part: " + part);
                 }
             }
-            if (parts.length > 1 && parts[1].trim().equals("<none>")) {
+            if (parts.length > 1 && parts[1].trim().equals(NONE) && 
+                    SESSION_TYPE_DESKTOP.equals(parts[2])) {
                 targetIP = parts[0].trim();
```

Comment:
I think it would have been better to add the sessionType to the selector (rather than get multiple results back).  Maybe in another PR you could try that change.  Otherwise, looks good.  Thanks.